index.js

```diff
@@ -1,24 +1,65 @@
 'use strict';
 const pathExists = require('path-exists');
 const modifyFilename = require('modify-filename');
+const slash = require('slash');
+
+const defaultIncrementer = (filename, extension, counter) =>
+	`${filename} (${counter})${extension}`;
+
+const findUnusedFilePath = (filePath, options, sync) => {
+	options = {
+		incrementer: defaultIncrementer,
+		...options
+	};
 
-const incrementer = filePath => {
 	let counter = 0;
-	return () => modifyFilename(filePath, (filename, extension) => `${filename} (${++counter})${extension}`);
-};
+	const triedFileNames = new Set();
+
+	const incrementer = () =>
+		modifyFilename(filePath, (filename, extension) =>
+			options.incrementer(filename, extension, ++counter)
+		);
+
+	const checkIsTried = newFilePath => {
+		if (triedFileNames.has(newFilePath)) {
+			throw new Error('`incrementer` should return unique names');
```

Comment:
I don't know, should it? 

If I'm generating a random 4-characters name (like `U1HA`), but it's already taken and my incrementer generates another random 4-characters name it may happen that it generates `U1HA` again, I don't think the app should crash in this scenario.

Also maybe a name we tried earlier on it's free to use now, I don't think this check is necessary.

Maybe a `maxTries` options should be added instead, so if for some reason there are no available names the library throws an error after `maxTries` number of tries rather than being busy forever.

Reply:
What do you think, @sindresorhus

Reply:
I agree, it should not throw. As for `maxTries`, I cannot think of a sensible default, so I would go with `Infinity` (Unless @fabiospampinato can argument for something better).

Reply:
I'm setting it to [128](https://github.com/fabiospampinato/get-unused-path/blob/7d066c1b528e5bf0f5a7c7f9a5466367a1e659cd/src/index.ts#L15), but it's just a number I pulled out of nowhere, my reasoning against `Infinity` was that if it's impossible to find an unused path for whatever reason I don't want the app to hang forever, you don't want it to give up trying too early either though, so I'm setting it to 128 by default which feels about right to me.